File: src/main/kotlin/jimin/67week/테이블 해시 함수.py
def solution(data, col, row_begin, row_end):
    answer = 0
    data.sort(key=lambda x: (x[col - 1], -x[0]))

    for i in range(len(data[0])):
        answer += data[row_begin - 1][i] % row_begin

    for i in range(row_begin, row_end):
        sumi = 0
        for j in range(len(data[0])):
            sumi += data[i][j] % (i + 1)
        answer = answer ^ sumi

    return answer

File: src/main/kotlin/byeonghee/week67/테이블 해시 함수.kt
package byeonghee.week67

class 소병희_테이블해시함수 {
    fun solution(data: Array<IntArray>, col: Int, row_begin: Int, row_end: Int): Int {
        var answer = 0

        data.sortWith(compareBy<IntArray> { it[col-1] }.thenByDescending { it[0] })

        for(i in row_begin .. row_end) {
            var si = 0
            for(num in data[i-1]) {
                si += num % i
            }

            answer = answer xor si
        }

        return answer
    }
}